packages/schema/src/model.ts

```diff
@@ -58,24 +58,128 @@ export class JupyterGISModel implements IJupyterGISModel {
     this.annotationModel = annotationModel;
     this.settingRegistry = settingRegistry;
     this._pathChanged = new Signal<JupyterGISModel, string>(this);
+    this._settingsChanged = new Signal<JupyterGISModel, string>(this);
   }
 
   /**
    * Initialize custom settings for JupyterLab.
    */
-  async initSettings(): Promise<void> {
+  async initSettings() {
     if (this.settingRegistry) {
-      const setting = await this.settingRegistry.load(SETTINGS_ID);
-      this._settings = setting.composite as any;
+      try {
+        const setting = await this.settingRegistry.load(SETTINGS_ID);
+        this._settings = setting;
+
+        this._updateLocalSettings();
+
+        setting.changed.connect(() => {
```

Comment:
I'm not sure you understood the previous suggestion.

What I meant is that this `settings.changed.connect(() => {` should be able to take parameters: `settings.changed.connect((sender, args) => {`.

You can ignore sender, but you should log `args` and see what it contains. IIRC it should contain the settings key that has changed, so you can be more granular. 
Your current implementation still emits all events.

Reply:
My bad, your code is fine. My comment is invalid.